django/core/checks/messages.py

```diff
@@ -48,7 +52,16 @@ def is_serious(self, level=ERROR):
 
     def is_silenced(self):
         from django.conf import settings
-        return self.id in settings.SILENCED_SYSTEM_CHECKS
+
+        if self.id is None:
+            return False
```

Comment:
Fixing an edge case where `SILENCED_SYSTEM_CHECKS = [None]` would silence any number of checks that don't have id's.